submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,88 @@
+![Compacted 8-8 universal](small_8-8_compressed.jpg)
+
+The whole thing is unrecognizable, but that's because the secondary balancer is now inline with the main one, meaning we have to deal with up to 12 belts shoved into a 8 tile wide space, which forces the secondary balancer to be extremely spread out. When we downgrade from blue belts to red belts, we can also ditch the belt weaving entirely, and just use a couple blue undergrounds to get 8 red belts into a space of 4. This also makes it easy to downgrade to yellows, granted that we still have to use some red and blue undergrounds:
+
+![A blue, red and yellow universal 8-8s together](red_and_yellow_compressed.jpg)
+
+It's definitely possible to make a yellow belt design without any blue undergrounds, but that would make the whole thing much longer. It is also possible to make both the red and yellow designs without any higher-level undergrounds at all, but that would make the whole thing 33% wider and who knows how much longer.
+
+### Final explanations
+
+I intentionally left some mysteries in this article, as some were too awkward to explain in the process, and some I don't understand that well myself. This is the part where I (try to) clear them up, so fair warning that some technical stuff is coming.
+
+* Why is a balancer the most splitter-efficient redistributor?
+
+Let's prove this using induction. First, imagine that we already have the most efficient possible distributor across 2<sup>N-1</sup> belts, we'll call it the "small distributor". Now, let's add a row of splitters to split each of its outputs into 2. We now have a 2<sup>N-1</sup>-2<sup>N</sup> distributor. To have the right number of inputs, let's clone the small distributor, and feed its outputs to the remaining inputs of the added row of splitters. We now have a 2<sup>N</sup>-2<sup>N</sup> distributor with the smallest possible amount of splitters. Now we just say that a 2-2 distributor is a single splitter, which is obviously the most efficient. This efficient redistributor generation algorithm is identical to the simplest possible balancer generation algorithm, and you can feel free to test that yourself.
+
+* How do you guarantee input balancing?
+
+For any N-N balancer, if no outputs are congested, then all of them have the same throughput regardless of inputs. Similarly, if no inputs are starved, all of them have the same throughput regardless of outputs. So long as the loopback has enough capacity to feed all the missing inputs, there will be input balancing.
+
+* Why 3 less belts in the loopback?
+
+To be honest, I don't fully understand this myself. It's an experimentally obtained result, and it has something to do with 3 being the smallest number that doesn't divide 2<sup>N</sup>, which makes 3 open outputs the worst case the loopback has to deal with. I can't explain why 1 and 2 don't need full loopback throughput, though. If anyone wants to volunteer and help me out with the math here, I'd be more than grateful.
```

Comment:
You still need a conclusion. Copied from my first review:

You should end off just reiterating what you created. And then explain what the reader should do with this information. For example: Should he use it in his base? Should he try to design his own? Is it more a thought experiment? If you want to use it in your base, in what situation is it useful? Has this design been done before for an 8-8?